### PR TITLE
[Docs] Add docs for dropping table feature

### PR DESCRIPTION
**Pull request by cstavr**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [X] Other (docs)

## Description

Update documentation to cover `ALTER TABLE <table-name> DROP FEATURE <feature-name>` command.

## How was this patch tested?

N/A

## Does this PR introduce _any_ user-facing changes?

No.

### CODE REVIEW

File: docs/source/delta-deletion-vectors.md [hunk 1]
```diff
@@ -26,7 +26,10 @@ You enable support for deletion vectors on a <Delta> table by setting a <Delta>
 ALTER TABLE <table_name> SET TBLPROPERTIES ('delta.enableDeletionVectors' = true);
 ```
 
-.. warning:: When you enable deletion vectors, the table protocol version is upgraded. Table protocol version upgrades are not reversible. After upgrading, the table will not be readable by <Delta> clients that do not support deletion vectors. See [_](versioning.md).
+.. warning::
+  When you enable deletion vectors, the table protocol version is upgraded. After upgrading, the table will not be readable by <Delta> clients that do not support deletion vectors. See [_](versioning.md).
+
+  In <Delta> 3.1 and above, you can drop the deletion vectors table feature to enable compatibility with other Delta clients. See [_](delta-drop-feature.md).
```
Comment (vkorukanti) [inline docs/source/delta-deletion-vectors.md:32]: Is drop dv feature available from 3.0?

File: docs/source/delta-drop-feature.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+---
+description: Learn how to drop table features in <Delta> to downgrade reader and writer protocol requirements and resolve compatibility issues.
+oprhan: 1
+---
+
+# Drop Delta table features
+
+.. note:: This feature is available in <Delta> 3.1.0 and above.
```
Comment (vkorukanti) [inline docs/source/delta-drop-feature.md:8]: same comment.

**Review by vkorukanti — APPROVED** [review]
lgtm. Please make sure the doc contents are rendered correctly.

File: docs/source/delta-deletion-vectors.md (hunk 1 above)
Comment (cstavr) [reply docs/source/delta-deletion-vectors.md]: Correct. Fixed.

File: docs/source/delta-drop-feature.md (hunk 2 above)
Comment (cstavr) [reply docs/source/delta-drop-feature.md]: Done

**Comment by cstavr** [comment]
> lgtm. Please make sure the doc contents are rendered correctly.

Checked rendering and looks good.

**Pull request merged**